src/windows/models/properties_model.py

```diff
@@ -436,8 +440,12 @@ def value_updated(self, item, interpolation=-1, value=None, interpolation_detail
                     # Loop through points, find a matching points on this frame
                     found_point = False
                     point_to_delete = None
+                    last_point_idx = 0
                     for point in c.data[property_key]["Points"]:
                         log.info("looping points: co.X = %s" % point["co"]["X"])
+                        if point["co"]["X"] < self.frame_number:
+                            # Remember index(+1) of the point previous to the current frame
+                            last_point_idx += 1
```

Comment:
And, in fact, your code as written here won't work correctly, because it assumes the list _is_ already sorted. If the list is **not** sorted, like for instance my example `it` from our conversation in the comments, then you get this:

```python3
>>> pprint(it)
{'has_video': {'Points': [{'co': {'X': 1.0, 'Y': -1.0}, 'interpolation': 2},
                          {'co': {'X': 4, 'Y': 1}, 'interpolation': 1},
                          {'co': {'X': 9, 'Y': 0}, 'interpolation': 1},
                          {'co': {'X': 2, 'Y': 3}, 'interpolation': 1},
                          {'co': {'X': 47, 'Y': 13}, 'interpolation': 1},
                          {'co': {'X': 3, 'Y': 99}, 'interpolation': 1},
                          {'co': {'X': 3000, 'Y': 1}, 'interpolation': 1},
                          {'co': {'X': 17, 'Y': 45}, 'interpolation': 1},
                          {'co': {'X': 555, 'Y': 55}, 'interpolation': 1},
                          {'co': {'X': 554, 'Y': 54}, 'interpolation': 1},
                          {'co': {'X': 553, 'Y': 53}, 'interpolation': 1},
                          {'co': {'X': 552, 'Y': 52}, 'interpolation': 1},
                          {'co': {'X': 551, 'Y': 51}, 'interpolation': 1},
                          {'co': {'X': 550, 'Y': 50}, 'interpolation': 1},
                          {'co': {'X': 556, 'Y': 56}, 'interpolation': 1},
                          {'co': {'X': 558, 'Y': 58}, 'interpolation': 1},
                          {'co': {'X': 557, 'Y': 57}, 'interpolation': 1},
                          {'co': {'X': 549, 'Y': 49}, 'interpolation': 1},
                          {'co': {'X': 6, 'Y': 0}, 'interpolation': 1},
                          {'co': {'X': 2999, 'Y': 12}, 'interpolation': 1}]}}
>>> idx = 0
>>> for point in it['has_video']['Points']:
...     if point['co']['X'] < 27:
...         idx += 1
... 
>>> print(idx)
7
```

The correct way to keep track of your current list index when looping over a list in Python is with `enumerate()`, e.g.:
```python3
>>> for (i,  point) in enumerate(it['has_video']['Points']):
...     if point['co']['X'] == 553: print("At index {}".format(i))
... 
At index 10
```

...But that would still require the list to be already-sorted, if you wanted to insert into the correct location. If it's not sorted, there _is_ no "correct location".

Reply:
In fact, the existing code really shouldn't be looping over the list _anyway_, just to search for a row. That's inefficient. A more Pythonic method would be to use `filter()` to find an existing element:

```python3
>>> for row in list(filter(lambda i: i['co']['X'] == 557, arr)):
...     print(row)
... 
{'co': {'X': 557, 'Y': 57}, 'interpolation': 1}
>>> for row in list(filter(lambda i: i['co']['X'] == 510, arr)):
...     print(row)
... 
>>>
```